# GIR 온실가스 배출량 분석

GIR 데이터를 활용해 국내 기업의 온실가스 배출량과 에너지 사용량을 확인하고, 이후 KCGS 환경등급과 실제 환경성과의 관계를 분석하려고 한다.

먼저 2020~2025년 데이터의 구조와 연도 정보를 확인했다.

In [1]:
import pandas as pd
from pathlib import Path

# GIR 데이터 파일 경로
data_path = Path("../data")

gir_files = sorted(data_path.glob("gir_*.csv"))

gir_files

[WindowsPath('../data/gir_2020.csv'),
 WindowsPath('../data/gir_2021.csv'),
 WindowsPath('../data/gir_2022.csv'),
 WindowsPath('../data/gir_2023.csv'),
 WindowsPath('../data/gir_2024.csv'),
 WindowsPath('../data/gir_2025.csv'),
 WindowsPath('../data/gir_company_2020_2025.csv')]

In [2]:
# 파일명과 실제 대상년도 확인
for file in gir_files:
    df = pd.read_csv(file)

    print(
        file.name,
        '행 수:', len(df),
        '대상년도:', df['대상년도'].unique()
    )

gir_2020.csv 행 수: 1061 대상년도: [2020]
gir_2021.csv 행 수: 1079 대상년도: [2021]
gir_2022.csv 행 수: 1102 대상년도: [2022]
gir_2023.csv 행 수: 1540 대상년도: [2023]
gir_2024.csv 행 수: 1167 대상년도: [2024]
gir_2025.csv 행 수: 1173 대상년도: [2025]
gir_company_2020_2025.csv 행 수: 6713 대상년도: [2020 2021 2022 2023 2024 2025]


In [3]:
# 연도별 지정구분 확인
for file in gir_files:
    df = pd.read_csv(file)

    print(f'\n{file.name}')
    print(df['지정구분'].value_counts(dropna=False))


gir_2020.csv
지정구분
사업장    563
업체     498
Name: count, dtype: int64

gir_2021.csv
지정구분
사업장    576
업체     503
Name: count, dtype: int64

gir_2022.csv
지정구분
사업장    718
업체     384
Name: count, dtype: int64

gir_2023.csv
지정구분
사업장    1101
업체      439
Name: count, dtype: int64

gir_2024.csv
지정구분
사업장    796
업체     371
Name: count, dtype: int64

gir_2025.csv
지정구분
사업장    796
업체     377
Name: count, dtype: int64

gir_company_2020_2025.csv
지정구분
사업장    4194
업체     2519
Name: count, dtype: int64


In [4]:
# 2023년 데이터 확인
gir_2023 = pd.read_csv("../data/gir_2023.csv")

# 완전히 동일한 중복 행 수
print(
    '2023년 완전 중복 행 수:',
    gir_2023.duplicated().sum()
)

2023년 완전 중복 행 수: 0


In [5]:
# 2023년 완전히 동일한 중복 행 수 확인
print(
    '2023년 완전 중복 행 수:',
    gir_2023.duplicated().sum()
)

2023년 완전 중복 행 수: 0


In [6]:
# 연도별 완전 중복 행 수 확인
for file in gir_files:
    df = pd.read_csv(file)

    print(
        file.name,
        '완전 중복:',
        df.duplicated().sum()
    )

gir_2020.csv 완전 중복: 0
gir_2021.csv 완전 중복: 0
gir_2022.csv 완전 중복: 0
gir_2023.csv 완전 중복: 0
gir_2024.csv 완전 중복: 0
gir_2025.csv 완전 중복: 0
gir_company_2020_2025.csv 완전 중복: 0


In [7]:
# 연도별 법인명 중복 확인
for file in gir_files:
    df = pd.read_csv(file)

    duplicated_company = (
        df['법인명']
        .value_counts()
    )

    print(f'\n{file.name}')
    print('전체 행 수:', len(df))
    print('고유 법인 수:', df['법인명'].nunique())
    print('2번 이상 등장한 법인 수:', (duplicated_company >= 2).sum())


gir_2020.csv
전체 행 수: 1061
고유 법인 수: 1060
2번 이상 등장한 법인 수: 1

gir_2021.csv
전체 행 수: 1079
고유 법인 수: 1078
2번 이상 등장한 법인 수: 1

gir_2022.csv
전체 행 수: 1102
고유 법인 수: 1095
2번 이상 등장한 법인 수: 7

gir_2023.csv
전체 행 수: 1540
고유 법인 수: 1142
2번 이상 등장한 법인 수: 398

gir_2024.csv
전체 행 수: 1167
고유 법인 수: 1166
2번 이상 등장한 법인 수: 1

gir_2025.csv
전체 행 수: 1173
고유 법인 수: 1172
2번 이상 등장한 법인 수: 1

gir_company_2020_2025.csv
전체 행 수: 6713
고유 법인 수: 1548
2번 이상 등장한 법인 수: 1422


In [8]:
# 2023년에 여러 번 등장한 법인 확인
company_count_2023 = (
    gir_2023['법인명']
    .value_counts()
)

duplicate_companies_2023 = company_count_2023[
    company_count_2023 >= 2
].index

gir_2023[
    gir_2023['법인명'].isin(duplicate_companies_2023)
][
    [
        '법인명',
        '지정구분',
        '온실가스 배출량(tCO₂eq)',
        '에너지 사용량(TJ)'
    ]
].sort_values('법인명').head(30)

,법인명,지정구분,온실가스 배출량(tCO₂eq),에너지 사용량(TJ)
0,(유)남해환경,사업장,29884,196
1,(유)남해환경,사업장,29884,196
3,(유)청명,사업장,19665,392
4,(유)청명,사업장,19665,392
5,(유)호남고속,사업장,21201,327
6,(유)호남고속,사업장,21201,327
7,(재)우체국물류지원단,사업장,18259,269
8,(재)우체국물류지원단,사업장,18259,269
17,(주)경동원,사업장,8281,163
18,(주)경동원,사업장,8281,163


In [9]:
# 2023년 데이터 컬럼 확인
gir_2023.columns

Index(['번호', '관장기관', '법인명', '대상년도', '지정구분', '지정업종(목표)/<br />계획업종(할당)',
       '온실가스 배출량(tCO₂eq)', '에너지 사용량(TJ)', '검증수행기관', '비고'],
      dtype='str')

In [10]:
# 법인이 최대 몇 번까지 반복되는지 확인
gir_2023['법인명'].value_counts().value_counts().sort_index()

count
1    744
2    398
Name: count, dtype: int64

In [11]:
# 실제 컬럼 중 '번호'만 제외
check_cols = [
    col for col in gir_2023.columns
    if col != '번호'
]

# 번호만 다르고 나머지 내용이 같은 행 확인
print(
    '번호 제외 중복 행 수:',
    gir_2023.duplicated(
        subset=check_cols
    ).sum()
)

번호 제외 중복 행 수: 386


In [12]:
# 번호만 다르고 나머지 내용이 같은 중복을 임시 제거
gir_2023_temp = gir_2023.drop_duplicates(
    subset=check_cols,
    keep='first'
).copy()

print('중복 제거 전:', len(gir_2023))
print('중복 제거 후:', len(gir_2023_temp))
print('고유 법인 수:', gir_2023_temp['법인명'].nunique())

중복 제거 전: 1540
중복 제거 후: 1154
고유 법인 수: 1142


In [13]:
# 중복 제거 후에도 여러 번 등장하는 법인 찾기
remaining_counts = gir_2023_temp['법인명'].value_counts()

remaining_names = remaining_counts[
    remaining_counts >= 2
].index

print('여전히 여러 번 등장하는 법인 수:', len(remaining_names))

# 실제 행 확인
gir_2023_temp[
    gir_2023_temp['법인명'].isin(remaining_names)
].sort_values('법인명')

여전히 여러 번 등장하는 법인 수: 12


,번호,관장기관,법인명,대상년도,지정구분,지정업종(목표)/<br />계획업종(할당),온실가스 배출량(tCO₂eq),에너지 사용량(TJ),검증수행기관,비고
163,164,산업통상자원부,(주)에스이그린에너지,2023,사업장,발전에너지,54777,0,한국표준협회,NaN
164,165,산업통상자원부,(주)에스이그린에너지,2023,업체,발전에너지,54777,0,한국표준협회,NaN
392,393,환경부,경원여객자동차(주),2023,사업장,육상 여객 운송업,29323,540,한국표준협회,소재지 : 경기도 안산
393,394,환경부,경원여객자동차(주),2023,사업장,육상 여객 운송업,15360,218,(주)한국경영인증원,소재지 : 경상남도 진주
482,483,산업통상자원부,노을그린에너지 주식회사,2023,업체,발전에너지,21018,4,(주)한국경영인증원,NaN
483,484,산업통상자원부,노을그린에너지 주식회사,2023,사업장,발전에너지,21018,4,(주)한국경영인증원,NaN
660,661,산업통상자원부,베페사징크코리아(주),2023,사업장,비철금속,112462,154,(재)한국화학융합시험연구원,NaN
661,662,산업통상자원부,베페사징크코리아(주),2023,업체,비철금속,112462,154,(재)한국화학융합시험연구원,NaN
683,684,해양수산부,부산신항만주식회사,2023,업체,항만,49973,988,한국표준협회,NaN
684,685,해양수산부,부산신항만주식회사,2023,사업장,항만,49973,988,한국표준협회,NaN


In [14]:
# 판단이 어려운 3개 기업
check_companies = [
    '에스케이피유코어 주식회사',
    '에이치디씨폴리올(주)',
    '쿠팡로지스틱스서비스 유한회사'
]

# 2022~2024 데이터 불러오기
gir_2022 = pd.read_csv("../data/gir_2022.csv")
gir_2024 = pd.read_csv("../data/gir_2024.csv")

# 세 연도 합치기
check_years = pd.concat(
    [gir_2022, gir_2023, gir_2024],
    ignore_index=True
)

# 해당 기업만 확인
check_years[
    check_years['법인명'].isin(check_companies)
][
    [
        '법인명',
        '대상년도',
        '지정구분',
        '온실가스 배출량(tCO₂eq)',
        '에너지 사용량(TJ)',
        '비고'
    ]
].sort_values(['법인명', '대상년도'])

,법인명,대상년도,지정구분,온실가스 배출량(tCO₂eq),에너지 사용량(TJ),비고
641,에스케이피유코어 주식회사,2022,사업장,24362,534,"""미쓰이케미칼앤드에스케이씨폴리우레탄 주식회사"" 업체명 변경"
2011,에스케이피유코어 주식회사,2023,사업장,22558,496,NaN
2012,에스케이피유코어 주식회사,2023,사업장,22556,496,NaN
2033,에이치디씨폴리올(주),2023,사업장,35086,588,NaN
2034,에이치디씨폴리올(주),2023,사업장,35085,588,NaN
3331,에이치디씨폴리올(주),2024,사업장,"33,360",565,NaN
2413,쿠팡로지스틱스서비스 유한회사,2023,사업장,94808,1531,NaN
2414,쿠팡로지스틱스서비스 유한회사,2023,사업장,94998,1535,NaN
3633,쿠팡로지스틱스서비스 유한회사,2024,사업장,"97,956","1,664",NaN


### 데이터 확인

2023년 데이터에서 동일 법인이 반복된 사례를 확인한 결과, 서로 다른 사업장이 실제로 존재하는 경우와 거의 동일한 값이 중복된 경우가 함께 나타났다.

기업 단위 분석에서는 `업체` 행이 있는 경우 업체 값을 우선 사용하고, 업체 행 없이 서로 다른 사업장이 명확한 경우에는 사업장 값을 합산하기로 했다.

반면 동일 사업장으로 보이지만 배출량이 거의 같은 값으로 반복된 일부 기업은 어떤 값이 최종값인지 확인하기 어려워 2023년 기업 단위 분석에서 제외하기로 했다.

In [15]:
# 배출량과 에너지 사용량을 숫자로 변환
numeric_cols = [
    '온실가스 배출량(tCO₂eq)',
    '에너지 사용량(TJ)'
]

gir_2023_temp = gir_2023.copy()

for col in numeric_cols:
    gir_2023_temp[col] = pd.to_numeric(
        gir_2023_temp[col]
        .astype(str)
        .str.replace(',', '', regex=False),
        errors='coerce'
    )


# 번호만 다른 완전 중복 제거
check_cols = [
    col for col in gir_2023_temp.columns
    if col != '번호'
]

gir_2023_temp = gir_2023_temp.drop_duplicates(
    subset=check_cols,
    keep='first'
).copy()


# 값이 거의 같지만 어느 행이 최종값인지 판단하기 어려운 기업은 제외
ambiguous_2023 = [
    '에스케이피유코어 주식회사',
    '에이치디씨폴리올(주)',
    '쿠팡로지스틱스서비스 유한회사'
]

gir_2023_temp = gir_2023_temp[
    ~gir_2023_temp['법인명'].isin(ambiguous_2023)
].copy()

In [16]:
company_rows = []

for company, group in gir_2023_temp.groupby('법인명'):

    # 같은 내용인데 비고만 다른 행은 한 번만 사용
    core_cols = [
        col for col in group.columns
        if col not in ['번호', '비고']
    ]

    group = group.drop_duplicates(
        subset=core_cols,
        keep='first'
    )

    # 업체 행이 있는 경우
    company_level = group[
        group['지정구분'] == '업체'
    ]

    if len(company_level) > 0:

        # 기업 단위 분석에서는 업체 값을 우선 사용
        row = company_level.iloc[0].copy()
        row['집계방식'] = '업체 우선'

    else:

        # 업체 행이 없으면 사업장 데이터 사용
        sites = group[
            group['지정구분'] == '사업장'
        ]

        if len(sites) == 1:

            row = sites.iloc[0].copy()
            row['집계방식'] = '단일 사업장'

        else:

            # 여러 사업장이 실제로 존재하면 합산
            row = sites.iloc[0].copy()

            row['온실가스 배출량(tCO₂eq)'] = (
                sites['온실가스 배출량(tCO₂eq)'].sum()
            )

            row['에너지 사용량(TJ)'] = (
                sites['에너지 사용량(TJ)'].sum()
            )

            row['집계방식'] = f'사업장 {len(sites)}개 합산'

    company_rows.append(row)


# 기업 단위 데이터 생성
gir_2023_company = pd.DataFrame(company_rows)

gir_2023_company.shape

(1139, 11)

In [17]:
print('기업 단위 행 수:', len(gir_2023_company))
print('고유 법인 수:', gir_2023_company['법인명'].nunique())

print('\n집계방식')
print(gir_2023_company['집계방식'].value_counts())

기업 단위 행 수: 1139
고유 법인 수: 1139

집계방식
집계방식
단일 사업장       748
업체 우선        390
사업장 2개 합산      1
Name: count, dtype: int64


In [18]:
# 번호만 다른 실질 중복 먼저 제거
check_cols = [
    col for col in gir_2023.columns
    if col != '번호'
]

gir_2023_clean = (
    gir_2023
    .drop_duplicates(subset=check_cols, keep='first')
    .copy()
)

# 업체 행만 추출
company_type_rows = gir_2023_clean[
    gir_2023_clean['지정구분'] == '업체'
]

# 같은 법인에 업체 행이 여러 개 있는지 확인
company_type_count = (
    company_type_rows['법인명']
    .value_counts()
)

company_type_count[
    company_type_count >= 2
]

Series([], Name: count, dtype: int64)

In [19]:
# 숫자형 컬럼 정리
numeric_cols = [
    '온실가스 배출량(tCO₂eq)',
    '에너지 사용량(TJ)'
]

for col in numeric_cols:
    gir_2023_clean[col] = pd.to_numeric(
        gir_2023_clean[col]
        .astype(str)
        .str.replace(',', '', regex=False),
        errors='coerce'
    )


# 번호와 비고만 다르고 실제 내용이 같은 행 제거
core_cols = [
    col for col in gir_2023_clean.columns
    if col not in ['번호', '비고']
]

gir_2023_clean = gir_2023_clean.drop_duplicates(
    subset=core_cols,
    keep='first'
).copy()


# 최종값 판단이 어려운 3개 기업 제외
ambiguous_2023 = [
    '에스케이피유코어 주식회사',
    '에이치디씨폴리올(주)',
    '쿠팡로지스틱스서비스 유한회사'
]

gir_2023_clean = gir_2023_clean[
    ~gir_2023_clean['법인명'].isin(ambiguous_2023)
].copy()

In [20]:
company_rows = []

for company, group in gir_2023_clean.groupby('법인명'):

    # 업체 행이 있으면 기업 전체 값으로 우선 사용
    company_row = group[
        group['지정구분'] == '업체'
    ]

    if len(company_row) == 1:
        row = company_row.iloc[0].copy()
        row['집계방식'] = '업체 우선'

    else:
        # 업체 행이 없으면 사업장 사용
        sites = group[
            group['지정구분'] == '사업장'
        ]

        if len(sites) == 1:
            row = sites.iloc[0].copy()
            row['집계방식'] = '단일 사업장'

        else:
            # 실제 여러 사업장이 있으면 합산
            row = sites.iloc[0].copy()

            row['온실가스 배출량(tCO₂eq)'] = (
                sites['온실가스 배출량(tCO₂eq)'].sum()
            )

            row['에너지 사용량(TJ)'] = (
                sites['에너지 사용량(TJ)'].sum()
            )

            row['집계방식'] = f'사업장 {len(sites)}개 합산'

    company_rows.append(row)


gir_2023_company = pd.DataFrame(company_rows)

In [21]:
print('기업 단위 행 수:', len(gir_2023_company))
print('고유 법인 수:', gir_2023_company['법인명'].nunique())

print('\n집계방식')
print(gir_2023_company['집계방식'].value_counts())

기업 단위 행 수: 1139
고유 법인 수: 1139

집계방식
집계방식
단일 사업장       748
업체 우선        390
사업장 2개 합산      1
Name: count, dtype: int64


In [22]:
# 2020~2025년 GIR 파일의 반복 법인 확인

for file in gir_files:
    df = pd.read_csv(file)

    # 번호와 비고를 제외한 실제 내용이 같은 행 제거
    check_cols = [
        col for col in df.columns
        if col not in ['번호', '비고']
    ]

    df_clean = df.drop_duplicates(
        subset=check_cols,
        keep='first'
    ).copy()

    # 두 번 이상 등장하는 법인
    company_count = df_clean['법인명'].value_counts()

    duplicate_names = company_count[
        company_count >= 2
    ].index

    print(f'\n===== {file.name} =====')
    print('반복 법인 수:', len(duplicate_names))

    if len(duplicate_names) > 0:
        display(
            df_clean[
                df_clean['법인명'].isin(duplicate_names)
            ][
                [
                    '법인명',
                    '지정구분',
                    '온실가스 배출량(tCO₂eq)',
                    '에너지 사용량(TJ)',
                    '비고'
                ]
            ].sort_values('법인명')
        )


===== gir_2020.csv =====
반복 법인 수: 1


,법인명,지정구분,온실가스 배출량(tCO₂eq),에너지 사용량(TJ),비고
259,경원여객자동차(주),사업장,17922.0,253.0,NaN
260,경원여객자동차(주),사업장,35205.0,633.0,NaN



===== gir_2021.csv =====
반복 법인 수: 1


,법인명,지정구분,온실가스 배출량(tCO₂eq),에너지 사용량(TJ),비고
263,경원여객자동차(주),사업장,33477,603,NaN
264,경원여객자동차(주),사업장,14574,207,NaN



===== gir_2022.csv =====
반복 법인 수: 5


,법인명,지정구분,온실가스 배출량(tCO₂eq),에너지 사용량(TJ),비고
267,경원여객자동차(주),사업장,32392,595,NaN
268,경원여객자동차(주),사업장,14103,200,NaN
1090,효성첨단소재 주식회사,업체,178541,3999,NaN
1091,효성첨단소재 주식회사,업체,178541,3994,NaN
1092,효성티앤씨 주식회사,업체,297685,6842,NaN
1093,효성티앤씨 주식회사,업체,297685,6848,NaN
1094,효성화학 주식회사,업체,862233,18753,NaN
1095,효성화학 주식회사,업체,862233,18757,NaN
1098,휴비스,업체,384776,5891,NaN
1099,휴비스,업체,384776,5892,NaN



===== gir_2023.csv =====
반복 법인 수: 11


,법인명,지정구분,온실가스 배출량(tCO₂eq),에너지 사용량(TJ),비고
163,(주)에스이그린에너지,사업장,54777,0,NaN
164,(주)에스이그린에너지,업체,54777,0,NaN
392,경원여객자동차(주),사업장,29323,540,소재지 : 경기도 안산
393,경원여객자동차(주),사업장,15360,218,소재지 : 경상남도 진주
482,노을그린에너지 주식회사,업체,21018,4,NaN
483,노을그린에너지 주식회사,사업장,21018,4,NaN
660,베페사징크코리아(주),사업장,112462,154,NaN
661,베페사징크코리아(주),업체,112462,154,NaN
683,부산신항만주식회사,업체,49973,988,NaN
684,부산신항만주식회사,사업장,49973,988,NaN



===== gir_2024.csv =====
반복 법인 수: 1


,법인명,지정구분,온실가스 배출량(tCO₂eq),에너지 사용량(TJ),비고
275,경원여객자동차(주),사업장,"15,471",220,소재지 : 경상남도 진주
276,경원여객자동차(주),사업장,"31,036",572,소재지 : 경기도 안산



===== gir_2025.csv =====
반복 법인 수: 1


,법인명,지정구분,온실가스 배출량(tCO₂eq),에너지 사용량(TJ),비고
282,경원여객자동차(주),사업장,"15,574",221,소재지 : 경상남도 진주
283,경원여객자동차(주),사업장,"29,532",549,소재지 : 경기도 안산



===== gir_company_2020_2025.csv =====
반복 법인 수: 1422


,법인명,지정구분,온실가스 배출량(tCO₂eq),에너지 사용량(TJ),비고
4375,롯데이노베이트(주),사업장,120487.0,2514.0,NaN
5541,롯데이노베이트(주),사업장,123522.0,2578.0,NaN
4376,리뉴에너지경인 주식회사,사업장,51427.0,12.0,NaN
5542,리뉴에너지경인 주식회사,사업장,51995.0,102.0,NaN
4377,(유)남해환경,사업장,27986.0,178.0,NaN
...,...,...,...,...,...
2137,희성촉매(주),사업장,21146.0,423.0,NaN
4374,희성촉매(주),사업장,21525.0,439.0,NaN
5540,희성촉매(주),사업장,23978.0,488.0,NaN
3232,희성촉매(주),사업장,21319.0,428.0,NaN


In [23]:
# GIR 데이터를 기업 단위로 정리하는 함수
def make_company_data(df):

    df = df.copy()

    # 숫자형 컬럼 정리
    numeric_cols = [
        '온실가스 배출량(tCO₂eq)',
        '에너지 사용량(TJ)'
    ]

    for col in numeric_cols:
        df[col] = pd.to_numeric(
            df[col]
            .astype(str)
            .str.replace(',', '', regex=False),
            errors='coerce'
        )

    # 번호와 비고만 다른 동일 데이터 제거
    check_cols = [
        col for col in df.columns
        if col not in ['번호', '비고']
    ]

    df = df.drop_duplicates(
        subset=check_cols,
        keep='first'
    ).copy()

    company_rows = []

    for company, group in df.groupby('법인명'):

        company_level = group[
            group['지정구분'] == '업체'
        ]

        # 업체 행이 1개이면 그대로 사용
        if len(company_level) == 1:

            row = company_level.iloc[0].copy()
            row['집계방식'] = '업체 우선'

        # 업체 행이 여러 개인 경우
        elif len(company_level) > 1:

            ghg_values = company_level[
                '온실가스 배출량(tCO₂eq)'
            ].dropna().unique()

            # 배출량 값이 모두 같으면 한 번만 사용
            if len(ghg_values) == 1:

                row = company_level.iloc[0].copy()

                row['온실가스 배출량(tCO₂eq)'] = ghg_values[0]

                # 에너지 값이 서로 다르면 임의 선택하지 않음
                energy_values = company_level[
                    '에너지 사용량(TJ)'
                ].dropna().unique()

                if len(energy_values) > 1:
                    row['에너지 사용량(TJ)'] = pd.NA

                row['집계방식'] = '업체 중복값 확인'

            else:
                # 서로 다른 업체 값이면 판단하지 않고 제외
                continue

        else:

            sites = group[
                group['지정구분'] == '사업장'
            ]

            if len(sites) == 1:

                row = sites.iloc[0].copy()
                row['집계방식'] = '단일 사업장'

            else:

                # 사업장 여러 개면 일단 법인 단위로 합산
                row = sites.iloc[0].copy()

                row['온실가스 배출량(tCO₂eq)'] = (
                    sites['온실가스 배출량(tCO₂eq)'].sum()
                )

                row['에너지 사용량(TJ)'] = (
                    sites['에너지 사용량(TJ)'].sum()
                )

                row['집계방식'] = f'사업장 {len(sites)}개 합산'

        company_rows.append(row)

    return pd.DataFrame(company_rows)

In [24]:
gir_2022 = pd.read_csv("../data/gir_2022.csv")

gir_2022_company = make_company_data(gir_2022)

print('기업 단위 행 수:', len(gir_2022_company))
print('고유 법인 수:', gir_2022_company['법인명'].nunique())

print('\n집계방식')
print(gir_2022_company['집계방식'].value_counts())

기업 단위 행 수: 1095
고유 법인 수: 1095

집계방식
집계방식
단일 사업장       714
업체 우선        376
업체 중복값 확인      4
사업장 2개 합산      1
Name: count, dtype: int64


In [25]:
# 2023년 데이터에 같은 함수 적용
gir_2023 = pd.read_csv("../data/gir_2023.csv")

gir_2023_company_test = make_company_data(gir_2023)

print('기업 단위 행 수:', len(gir_2023_company_test))
print('고유 법인 수:', gir_2023_company_test['법인명'].nunique())

print('\n집계방식')
print(gir_2023_company_test['집계방식'].value_counts())

기업 단위 행 수: 1142
고유 법인 수: 1142

집계방식
집계방식
단일 사업장       748
업체 우선        390
사업장 2개 합산      4
Name: count, dtype: int64


In [26]:
# 2023년에 사업장 2개를 합산한 기업 확인
gir_2023_company_test[
    gir_2023_company_test['집계방식'] == '사업장 2개 합산'
][
    [
        '법인명',
        '온실가스 배출량(tCO₂eq)',
        '에너지 사용량(TJ)',
        '집계방식'
    ]
]

,법인명,온실가스 배출량(tCO₂eq),에너지 사용량(TJ),집계방식
392,경원여객자동차(주),44683.0,758.0,사업장 2개 합산
909,에스케이피유코어 주식회사,45114.0,992.0,사업장 2개 합산
931,에이치디씨폴리올(주),70171.0,1176.0,사업장 2개 합산
1311,쿠팡로지스틱스서비스 유한회사,189806.0,3066.0,사업장 2개 합산


In [27]:
# 2023년에 중복 여부를 명확하게 판단하기 어려운 기업
ambiguous_2023 = [
    '에스케이피유코어 주식회사',
    '에이치디씨폴리올(주)',
    '쿠팡로지스틱스서비스 유한회사'
]

# 해당 3개 기업 제외
gir_2023_safe = gir_2023[
    ~gir_2023['법인명'].isin(ambiguous_2023)
].copy()

# 앞에서 만든 함수로 기업 단위 데이터 생성
gir_2023_company = make_company_data(
    gir_2023_safe
)

print('기업 단위 행 수:', len(gir_2023_company))
print('고유 법인 수:', gir_2023_company['법인명'].nunique())

print('\n집계방식')
print(gir_2023_company['집계방식'].value_counts())

기업 단위 행 수: 1139
고유 법인 수: 1139

집계방식
집계방식
단일 사업장       748
업체 우선        390
사업장 2개 합산      1
Name: count, dtype: int64


In [28]:
gir_2023_company[
    gir_2023_company['집계방식'].str.contains(
        '사업장 2개',
        na=False
    )
][
    [
        '법인명',
        '온실가스 배출량(tCO₂eq)',
        '에너지 사용량(TJ)',
        '집계방식'
    ]
]

,법인명,온실가스 배출량(tCO₂eq),에너지 사용량(TJ),집계방식
392,경원여객자동차(주),44683.0,758.0,사업장 2개 합산


## 2020~2025년 기업 단위 GIR 데이터 통합

연도별 GIR 데이터를 기업 단위로 정리한 뒤 하나의 데이터로 통합했다.
2023년에 중복 여부를 명확하게 판단하기 어려웠던 일부 기업은 잘못된 배출량 합산을 방지하기 위해 제외했다.

In [29]:
# 2023년에 판단이 어려웠던 기업
ambiguous_2023 = [
    '에스케이피유코어 주식회사',
    '에이치디씨폴리올(주)',
    '쿠팡로지스틱스서비스 유한회사'
]

gir_company_list = []

for year in range(2020, 2026):

    # 연도별 데이터 불러오기
    df = pd.read_csv(
        f"../data/gir_{year}.csv"
    )

    # 2023년의 판단이 어려운 3개 기업만 제외
    if year == 2023:
        df = df[
            ~df['법인명'].isin(ambiguous_2023)
        ].copy()

    # 기업 단위 데이터로 정리
    company_df = make_company_data(df)

    gir_company_list.append(company_df)

    print(
        year,
        '기업 수:',
        len(company_df)
    )

2020 기업 수: 1060
2021 기업 수: 1078
2022 기업 수: 1095
2023 기업 수: 1139
2024 기업 수: 1166
2025 기업 수: 1172


In [30]:
# 2020~2025년 기업 단위 데이터 통합
gir_company = pd.concat(
    gir_company_list,
    ignore_index=True
)

print('전체 행 수:', len(gir_company))

# 연도별 기업 수 확인
gir_company[
    '대상년도'
].value_counts().sort_index()

전체 행 수: 6710


대상년도
2020    1060
2021    1078
2022    1095
2023    1139
2024    1166
2025    1172
Name: count, dtype: int64

In [31]:
# 한 기업이 같은 연도에 두 번 남아 있는지 확인
duplicate_company_year = gir_company.duplicated(
    subset=['법인명', '대상년도']
).sum()

print(
    '기업명 + 대상년도 중복:',
    duplicate_company_year
)

기업명 + 대상년도 중복: 0


## 온실가스 배출량 연도별 현황

기업 단위로 정리한 GIR 데이터를 활용해 2020~2025년 온실가스 배출량의 기본 분포를 확인했다.

연도마다 관리 대상 기업의 구성이 다를 수 있기 때문에 총배출량뿐만 아니라 기업 수와 중앙값도 함께 확인했다.

In [32]:
# 연도별 온실가스 배출량 기본 통계
ghg_year_summary = (
    gir_company
    .groupby('대상년도')
    .agg(
        기업수=('법인명', 'nunique'),
        총배출량=('온실가스 배출량(tCO₂eq)', 'sum'),
        평균배출량=('온실가스 배출량(tCO₂eq)', 'mean'),
        중앙값배출량=('온실가스 배출량(tCO₂eq)', 'median')
    )
    .round(1)
)

ghg_year_summary

,기업수,총배출량,평균배출량,중앙값배출량
대상년도,,,,
2020,1060,587517090.0,555309.2,43958.5
2021,1078,614837456.0,570879.7,51349.0
2022,1095,596467313.0,545216.9,52848.5
2023,1139,575863386.0,506031.1,53945.0
2024,1166,565618230.0,485509.2,51427.0
2025,1172,563672309.0,480949.1,50230.5


### 분석 결과

2020~2025년 GIR 데이터를 확인한 결과, 분석 대상 기업 수는 1,060개에서 1,172개로 증가했다.

총 온실가스 배출량은 2021년에 가장 높게 나타난 이후 지속적으로 감소했고, 기업당 평균 배출량도 같은 기간 감소하는 흐름을 보였다.

반면 중앙값 배출량은 2023년까지 증가한 뒤 2024년과 2025년에 다시 감소했다.

다만 연도마다 GIR에 포함된 기업 구성이 다르기 때문에, 이러한 변화만으로 동일 기업들의 온실가스 배출량이 감소했다고 판단하기는 어렵다. 따라서 다음 분석에서는 여러 연도에 공통으로 포함된 기업만 따로 비교하기로 했다.

In [33]:
# 기업별로 GIR 데이터에 등장한 연도 수 확인
company_year_count = (
    gir_company
    .groupby('법인명')['대상년도']
    .nunique()
)

# 2020~2025년 6개 연도 모두 존재하는 기업
common_companies = company_year_count[
    company_year_count == 6
].index

print('6개 연도 모두 존재하는 기업 수:', len(common_companies))

# 동일 기업 패널 데이터 생성
gir_balanced = gir_company[
    gir_company['법인명'].isin(common_companies)
].copy()

# 연도별 기업 수 확인
gir_balanced.groupby('대상년도')['법인명'].nunique()

6개 연도 모두 존재하는 기업 수: 742


대상년도
2020    742
2021    742
2022    742
2023    742
2024    742
2025    742
Name: 법인명, dtype: int64

## 동일 기업의 온실가스 배출량 변화

연도별 대상 기업 구성 차이의 영향을 줄이기 위해, 2020~2025년 6개 연도에 모두 포함된 기업만 따로 추출해 온실가스 배출량 변화를 비교했다.

In [34]:
# 6개 연도 모두 포함된 동일 기업들의 배출량 요약
balanced_summary = (
    gir_balanced
    .groupby('대상년도')
    .agg(
        기업수=('법인명', 'nunique'),
        총배출량=('온실가스 배출량(tCO₂eq)', 'sum'),
        평균배출량=('온실가스 배출량(tCO₂eq)', 'mean'),
        중앙값배출량=('온실가스 배출량(tCO₂eq)', 'median')
    )
    .round(1)
)

balanced_summary

,기업수,총배출량,평균배출량,중앙값배출량
대상년도,,,,
2020,742,539114305.0,728532.8,64567.0
2021,742,554469340.0,748271.7,70000.0
2022,742,530139963.0,715438.5,67703.0
2023,742,509529527.0,687624.2,67451.0
2024,742,496442234.0,669962.5,67159.0
2025,742,488451329.0,658290.2,65174.0


In [35]:
# 2020년 대비 2025년 총배출량 변화율
ghg_2020 = balanced_summary.loc[2020, '총배출량']
ghg_2025 = balanced_summary.loc[2025, '총배출량']

change_rate = (
    (ghg_2025 - ghg_2020) / ghg_2020 * 100
)

print(
    '2020년 대비 2025년 총배출량 변화율:',
    round(change_rate, 1),
    '%'
)

2020년 대비 2025년 총배출량 변화율: -9.4 %


### 분석 결과

2020~2025년 모든 연도에 공통으로 포함된 동일 기업만 비교한 결과, 2025년 총 온실가스 배출량은 2020년보다 9.4% 감소했다.

연도별 대상 기업 구성이 달라지는 영향을 줄인 뒤에도 배출량 감소가 나타났기 때문에, 전체 GIR 데이터에서 확인된 감소가 단순한 기업 구성 변화만으로 발생한 것은 아니었다.

다만 절대 배출량은 기업의 생산량이나 매출 규모 변화에도 영향을 받을 수 있기 때문에, 이를 곧바로 환경효율 개선으로 해석하지는 않았다.

In [36]:
# 기업별 2020년과 2025년 배출량 비교
company_change = (
    gir_balanced[
        gir_balanced['대상년도'].isin([2020, 2025])
    ]
    .pivot(
        index='법인명',
        columns='대상년도',
        values='온실가스 배출량(tCO₂eq)'
    )
    .dropna()
)

# 배출량 변화 계산
company_change['변화량'] = (
    company_change[2025] - company_change[2020]
)

# 증가 / 감소 / 동일 구분
company_change['변화방향'] = company_change['변화량'].apply(
    lambda x: '감소' if x < 0
    else '증가' if x > 0
    else '동일'
)

# 기업 수와 비율 확인
change_direction = pd.DataFrame({
    '기업수': company_change['변화방향'].value_counts(),
    '비율(%)': (
        company_change['변화방향']
        .value_counts(normalize=True)
        .mul(100)
        .round(1)
    )
})

change_direction

,기업수,비율(%)
변화방향,,
감소,379,51.2
증가,360,48.6
동일,1,0.1


### 분석 결과

2020년과 2025년의 동일 기업을 비교한 결과, 배출량이 감소한 기업은 51.2%, 증가한 기업은 48.6%로 기업 수 기준에서는 거의 비슷하게 나타났다.

따라서 전체 배출량이 9.4% 감소한 결과가 대부분의 기업에서 공통적으로 나타난 현상이라고 보기는 어렵다. 감소 기업의 감축 규모와 증가 기업의 증가 규모 차이가 전체 배출량 변화에 영향을 줬을 가능성이 있다.

In [37]:
# 감소 기업의 총 감소량
total_decrease = (
    company_change.loc[
        company_change['변화량'] < 0,
        '변화량'
    ].sum()
)

# 증가 기업의 총 증가량
total_increase = (
    company_change.loc[
        company_change['변화량'] > 0,
        '변화량'
    ].sum()
)

# 전체 순변화량
net_change = company_change['변화량'].sum()

print('감소 기업 총 변화량:', round(total_decrease))
print('증가 기업 총 변화량:', round(total_increase))
print('전체 순변화량:', round(net_change))

감소 기업 총 변화량: -71304718
증가 기업 총 변화량: 20516203
전체 순변화량: -50788515


### 분석 결과

배출량이 감소한 기업들의 총 감소량은 약 7,130만 tCO₂eq였고, 증가한 기업들의 총 증가량은 약 2,052만 tCO₂eq로 나타났다.

기업 수는 감소 기업과 증가 기업이 거의 비슷했지만, 감소 기업의 감축 규모가 더 크게 나타나면서 전체적으로 약 5,079만 tCO₂eq의 순감소가 발생했다.

따라서 전체 배출량 감소는 기업 수의 차이보다는 감소 기업과 증가 기업의 배출량 변화 규모 차이에서 나타난 것으로 볼 수 있다.


In [38]:
# 배출량 감소폭이 큰 기업부터 정렬
top_reducers = (
    company_change
    .sort_values('변화량')
    .head(10)
    .copy()
)

top_reducers

대상년도,2020,2025,변화량,변화방향
법인명,,,,
한국남동발전,42521216.0,30835303.0,-11685913.0,감소
한국동서발전 주식회사,34905756.0,26344859.0,-8560897.0,감소
한국중부발전,34774881.0,28063372.0,-6711509.0,감소
주식회사 포스코,75671169.0,69846022.0,-5825147.0,감소
한국남부발전 주식회사,29007394.0,24407208.0,-4600186.0,감소
(주)지에스동해전력,6797741.0,2693243.0,-4104498.0,감소
한국서부발전,30422589.0,27282477.0,-3140112.0,감소
주식회사 지에스이앤알,2057228.0,654.0,-2056574.0,감소
삼표시멘트,5986309.0,4269882.0,-1716427.0,감소


In [39]:
# 상위 10개 감소 기업의 총 감소량
top10_decrease = top_reducers['변화량'].sum()

# 전체 감소량 중 상위 10개 기업 비중
top10_share = (
    abs(top10_decrease) / abs(total_decrease) * 100
)

print(
    '상위 10개 감소 기업의 전체 감소량 비중:',
    round(top10_share, 1),
    '%'
)

상위 10개 감소 기업의 전체 감소량 비중: 69.8 %


### 분석 결과

배출량 감소폭이 큰 상위 10개 기업이 전체 감소량의 69.8%를 차지했다.

따라서 2020~2025년 동일 기업의 총배출량이 9.4% 감소했지만, 이러한 감소가 모든 기업에서 고르게 나타난 것은 아니었다. 전체 감소량의 상당 부분이 일부 대형 배출기업의 감축에서 발생한 것으로 나타났다.

기업 수 기준으로는 감소 기업과 증가 기업이 거의 비슷했기 때문에, 전체 배출량 감소를 국내 기업 전반의 공통적인 감축 흐름으로 확대해서 해석하지는 않았다.

In [40]:
# 배출량 감소폭이 큰 상위 10개 기업 확인
top_reducers_table = (
    company_change
    .sort_values('변화량')
    .head(10)
    .reset_index()
)

# 기업별 전체 감소량 기여도 계산
top_reducers_table['감소기여도(%)'] = (
    abs(top_reducers_table['변화량'])
    / abs(total_decrease)
    * 100
).round(1)

top_reducers_table[
    [
        '법인명',
        2020,
        2025,
        '변화량',
        '감소기여도(%)'
    ]
]

대상년도,법인명,2020,2025,변화량,감소기여도(%)
0,한국남동발전,42521216.0,30835303.0,-11685913.0,16.4
1,한국동서발전 주식회사,34905756.0,26344859.0,-8560897.0,12.0
2,한국중부발전,34774881.0,28063372.0,-6711509.0,9.4
3,주식회사 포스코,75671169.0,69846022.0,-5825147.0,8.2
4,한국남부발전 주식회사,29007394.0,24407208.0,-4600186.0,6.5
5,(주)지에스동해전력,6797741.0,2693243.0,-4104498.0,5.8
6,한국서부발전,30422589.0,27282477.0,-3140112.0,4.4
7,주식회사 지에스이앤알,2057228.0,654.0,-2056574.0,2.9
8,삼표시멘트,5986309.0,4269882.0,-1716427.0,2.4
9,엘지디스플레이(주),4748207.0,3347075.0,-1401132.0,2.0


In [41]:
# 상위 10개 감소 기업의 총 감소량
top10_decrease = top_reducers_table['변화량'].sum()

# 전체 감소량 중 상위 10개 기업 비중
top10_share = (
    abs(top10_decrease)
    / abs(total_decrease)
    * 100
)

print(
    '상위 10개 감소 기업의 전체 감소량 비중:',
    round(top10_share, 1),
    '%'
)

상위 10개 감소 기업의 전체 감소량 비중: 69.8 %


In [42]:
# 상위 10개 감소 기업의 총 감소량
top10_decrease = top_reducers_table['변화량'].sum()

# 전체 감소량 중 상위 10개 기업 비중
top10_share = (
    abs(top10_decrease)
    / abs(total_decrease)
    * 100
)

print(
    '상위 10개 감소 기업의 전체 감소량 비중:',
    round(top10_share, 1),
    '%'
)

상위 10개 감소 기업의 전체 감소량 비중: 69.8 %


In [43]:
# 상위 10개 감소 기업 이름
top10_names = top_reducers_table['법인명']

# 상위 10개 기업 제외
remaining_change = company_change[
    ~company_change.index.isin(top10_names)
].copy()

# 나머지 기업의 순변화량
remaining_net_change = remaining_change['변화량'].sum()

# 나머지 기업의 2020년 총배출량
remaining_2020 = remaining_change[2020].sum()

# 변화율
remaining_change_rate = (
    remaining_net_change
    / remaining_2020
    * 100
)

print('상위 10개 제외 기업 수:', len(remaining_change))
print('상위 10개 제외 순변화량:', round(remaining_net_change))
print('상위 10개 제외 변화율:', round(remaining_change_rate, 1), '%')

상위 10개 제외 기업 수: 730
상위 10개 제외 순변화량: -986120
상위 10개 제외 변화율: -0.4 %


In [44]:
# 상위 10개 감소 기업 이름
top10_names = top_reducers_table['법인명']

# 상위 10개 기업 제외
remaining_change = company_change[
    ~company_change.index.isin(top10_names)
].copy()

# 나머지 기업의 순변화량
remaining_net_change = remaining_change['변화량'].sum()

# 나머지 기업의 2020년 총배출량
remaining_2020 = remaining_change[2020].sum()

# 2020년 대비 2025년 변화율
remaining_change_rate = (
    remaining_net_change
    / remaining_2020
    * 100
)

print('상위 10개 제외 기업 수:', len(remaining_change))
print('상위 10개 제외 순변화량:', round(remaining_net_change))
print('상위 10개 제외 변화율:', round(remaining_change_rate, 1), '%')

상위 10개 제외 기업 수: 730
상위 10개 제외 순변화량: -986120
상위 10개 제외 변화율: -0.4 %


### GIR 온실가스 배출량 분석 결과

2020~2025년 GIR 데이터를 기업 단위로 정리한 뒤 온실가스 배출량 변화를 확인했다.

연도별 관리 대상 기업 구성이 달라지는 영향을 줄이기 위해 6개 연도에 모두 포함된 동일 기업만 비교한 결과, 2025년 총 온실가스 배출량은 2020년 대비 9.4% 감소했다.

하지만 기업별로 확인하면 배출량이 감소한 기업은 51.2%, 증가한 기업은 48.6%로 거의 비슷하게 나타났다. 감소 기업의 총 감소량은 약 7,130만 tCO₂eq였고, 증가 기업의 총 증가량은 약 2,052만 tCO₂eq로 감소 기업의 변화 규모가 더 컸다.

또한 배출량 감소폭이 큰 상위 10개 기업이 전체 감소량의 69.8%를 차지했다. 따라서 전체 배출량 감소가 모든 기업에서 고르게 나타난 흐름이라기보다 일부 대형 배출기업의 감축 영향이 크게 작용한 것으로 판단했다.

다만 절대 배출량은 기업 규모, 생산량, 업종 등의 영향을 받기 때문에 배출량 감소만으로 기업의 환경성과가 개선되었다고 단정하지 않았다. 이후 KCGS 환경등급 및 DART 재무데이터와 결합해 환경등급과 실제 배출성과, 배출집약도의 관계를 추가로 확인한다.